core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/localscan/LocalCsvFileScanOpDesc.java

```diff
@@ -86,17 +82,102 @@ public Schema sourceSchema() {
     }
 
     private Schema inferSchema(String headerLine) {
+        java.util.List<AttributeType> attributeType = IntStream
+                .range(0, headerLine.split(delimiter).length).mapToObj(i->AttributeType.INTEGER).collect(Collectors.toList());
+
+
+        try(BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            if (header != null && header)
+                reader.readLine();
+            int i = 0;
+            while((line = reader.readLine())!=null && i<=100) {
+                String[] records = line.split((delimiter));
+                attributeType = inferline(attributeType, records);
+                i++;
+            }
+        } catch (FileNotFoundException e) {
```

Comment:
combine all exceptions and throw `RuntimeException`.

Reply:
It seems like IOException must be included for FileReader
changed to
```
catch (RuntimeException | IOException e) {
            e.printStackTrace();
        }
```